### ISSUE TITLE
[BUG] describe index & data length mismatch - dask related

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#2094]
**Describe the bug**
We are having an index & data lengths mismatch. 

**Steps/Code to reproduce bug**
```python
import cudf
x = cudf.DataFrame({'x':[]})
y = x.describe()
/cudf/python/cudf/dataframe/numerical.py:261: RuntimeWarning: invalid value encountered in double_scalars
  return np.float64(self.sum(dtype=dtype)) / self.valid_count
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-12fa09856aa7> in <module>
      3 import numpy as np
      4 x = cudf.DataFrame({'x':[]})
----> 5 y = x.describe()

/cudf/python/cudf/dataframe/dataframe.py in describe(self, percentiles, include, exclude)
   2466         if not include and not exclude:
   2467             numeric_data = self.select_dtypes(np.number)
-> 2468             output_frame = _create_output_frame(numeric_data, percentiles)
   2469 
   2470         elif include == 'all':

/cudf/python/cudf/dataframe/dataframe.py in _create_output_frame(data, percentiles)
   2458             # hack because we don't support strings in indexes
   2459             columns = data.columns
-> 2460             out_df = data[columns[0]].describe(percentiles=percentiles)
   2461             for col in columns[1:]:
   2462                 out_df[col] = data[col].describe(percentiles=percentiles)[col]

/cudf/python/cudf/dataframe/series.py in describe(self, percentiles, include, exclude)
   1945 
   1946         if np.issubdtype(self.dtype, np.number):
-> 1947             return describe_numeric(self)
   1948         else:
   1949             raise NotImplementedError("Describing non-numeric columns is not "

/cudf/python/cudf/dataframe/series.py in describe_numeric(self)
   1931                 values_name = self.name
   1932 
-> 1933             return DataFrame({'stats': names, values_name: data})
   1934 
   1935         def describe_categorical(self):

/cudf/python/cudf/dataframe/dataframe.py in __init__(self, name_series, index)
    157                 name_series = name_series.items()
    158             for k, series in name_series:
--> 159                 self.add_column(k, series, forceindex=index is not None)
    160 
    161     def serialize(self, serialize):

/cudf/python/cudf/dataframe/dataframe.py in add_column(self, name, data, forceindex)
   1089         if isinstance(data, GeneratorType):
   1090             data = Series(data)
-> 1091         series = self._prepare_series_for_add(data, forceindex=forceindex)
   1092         series.name = name
   1093         self._cols[name] = series

/cudf/python/cudf/dataframe/dataframe.py in _prepare_series_for_add(self, col, forceindex)
   1059         series = Series(col) if not SCALAR else col
   1060         self._sanitize_columns(series)
-> 1061         series = self._sanitize_values(series, SCALAR)
   1062 
   1063         empty_index = len(self._index) == 0

/cudf/python/cudf/dataframe/dataframe.py in _sanitize_values(self, series, SCALAR)
   1039                 return Series(arr)
   1040         elif len(self) > 0 and len(sind) != len(index):
-> 1041             raise ValueError('Length of values does not match index length')
   1042         return series
   1043 

ValueError: Length of values does not match index length

```

Upon investigation the point of failure is `to_array` at this line: https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/dataframe/series.py#L1905

If we have `np.nan` or `None` values in cudf series, when it is being converted to an array they are being lost.

```python
print(cudf.Series([np.nan, 1, None, np.nan],dtype='float64').to_array())
[1.]
```

**Expected behavior**
Our behavior should be same as of what pandas `.values` is:
```python
print(pd.Series([np.nan, 1, None, np.nan], dtype='float64').values)
array([nan,  1., nan, nan])
```

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source(`branch-0.9`)

**Additional context**
This fails in cases when a partitions has empty dataframe and raises exception in dask. So as a fix we could replace `self.quantile(percentiles).to_array().tolist()` with `self.quantile(percentiles).astype('float64').to_pandas().to_list()`. But this would be a temporary one though only to unblock anyone using describe. We will have to fix our series to support `nan/None`

**Comment by kkraus14** [comment]
> ```python
> print(cudf.Series([np.nan, 1, None, np.nan],dtype='float64').to_array())
> [1.]
> ```

Those `np.nan` above will be converted into nulls which will then be removed in `to_array()`. If you pass `nan_as_null = False` to the Series call it should work as expected.

**Comment by kkraus14** [comment]
@galipremsagar In the `to_array()` call you can pass a `fillna='pandas'` parameter which will have it convert the nulls into `NaN` which will get converted back into nulls when read into the GPU.

**Comment by galipremsagar** [comment]
@kkraus14 yes, both the ways worked. Will fix it the `to_array` way. 👍

**Comment by kkraus14** [comment]
Fixed by #2139
